kubectl-cloudflow/cloudflowavro/avrocompatibility_test.go

```diff
@@ -0,0 +1,109 @@
+package cloudflowavro
+
+import (
+	"github.com/hamba/avro"
+	"github.com/stretchr/testify/assert"
+	"log"
+	"testing"
+)
+
+func Test_SimpleSchemaCompatibility(t *testing.T) {
+
+	readerSchema := `{
+	"type": "long",
+	"name": "data"
+}`
+
+	writerSchema := `{
+       "name": "data", 
+       "type": "int"
+}`
+
+	reader, err := avro.ParseWithCache(readerSchema, "", avro.DefaultSchemaCache)
+	writer, err := avro.ParseWithCache(writerSchema, "", avro.DefaultSchemaCache)
+	if err != nil {
+		log.Fatal(err)
+	}
+	s := avro.NewSchemaCompatibility()
+	assert.Equal(t, err, nil)
+	assert.Equal(t, s.Compatible(reader, writer), nil)
+	assert.Equal(t, CheckSchemaCompatibility(readerSchema, writerSchema), nil)
+}
+func Test_SchemaCompatibilityWithLogicalTypes(t *testing.T) {
+	var readerSchema1 = `{
+		"type" : "record",
+		"name": "test",
+		"fields" : [
+	{
+		"type": {
+			"type": "int",
+			"logicalType": "time-millis"
+		},
+		"name": "simple"
+	}] }`
+
+	var writerSchema1 = `
+	{
+	"type" : "record",
+	"name": "test",
+	"fields" : [
+	{
+	"name": "simple",
+	"type": "int"
+	}]}`
+
+	reader1, err := avro.ParseWithCache(readerSchema1, "", avro.DefaultSchemaCache)
+
+	assert.Equal(t, err, nil)
+
+	writer1, err := avro.ParseWithCache(writerSchema1, "", avro.DefaultSchemaCache)
+
+	assert.Equal(t, CheckSchemaCompatibility(readerSchema1, writerSchema1), nil)
+
+	assert.Equal(t, err, nil)
+
+	s := avro.NewSchemaCompatibility()
+
+	assert.Equal(t, s.Compatible(reader1, writer1), nil)
```

Comment:
I think I like it better if the Compatible returns bool

Reply:
Yes this is a relic of the past, I used to compare canonical schemas which then I found is not enough.